rotkehlchen/db/history_events.py

```diff
@@ -47,7 +47,8 @@ def add_history_event(    # pylint: disable=no-self-use
 
         May raise:
         - DeserializationError if the event could not be serialized for the DB
-        - sqlcipher.DatabaseError: If anything went wrong at insertion
+        - sqlcipher.IntegrityError: If the asset of the added history event does not exist in
+        the DB. Can only happen if an event with an unresolved asset is passed.
```

Comment:
There are places where we don't do this ccheck at the moment of building the history events like in https://github.com/rotki/rotki/blob/2560c73b54b68ec427b3b7af7776dfbaa1b59ec6/rotkehlchen/chain/ethereum/decoding/decoder.py#L363 I guess is user faults if a constant asset is deleted

Reply:
yes it is.